lib/flappy_search_bar.dart

```diff
@@ -340,37 +396,69 @@ class _SearchBarState<T> extends State<SearchBar<T>>
   }
 
   Widget _buildContent(BuildContext context) {
+    Widget content;
+
     if (_error != null) {
-      return _error;
+      content = _error;
     } else if (_loading) {
-      return widget.loader;
+      content = widget.loader;
     } else if (_searchQueryController.text.length < widget.minimumChars) {
-      if (widget.placeHolder != null) return widget.placeHolder;
-      return _buildListView(
+      if (widget.placeHolder != null) {
+        return widget.placeHolder;
+      }
+      content = _buildListView(
           widget.suggestions, widget.buildSuggestion ?? widget.onItemFound);
     } else if (_list.isNotEmpty) {
-      return _buildListView(_list, widget.onItemFound);
+      content = _buildListView(_list, widget.onItemFound);
     } else {
-      return widget.emptyWidget;
+      content = widget.centerEmptyWidget
+          ? Center(child: widget.emptyWidget)
+          : widget.emptyWidget;
     }
+
+    return Padding(
+      padding: widget.contentPadding,
+      child: content,
+    );
   }
 
   @override
   Widget build(BuildContext context) {
+    if (_firstLaunch && widget.autoFocus) {
+      /// Gives search bar focus on first build.
+      _firstLaunch = false;
+
+      final String currentQuery = _searchQueryController.text;
+
+      if (currentQuery.length >= widget.minimumChars) {
+        /// Runs an search immediately after the search bar is built.
+        searchBarController._search(
+          _searchQueryController.text,
+          widget.onSearch,
+        );
+      }
+
+      FocusScope.of(context).requestFocus(_searchQueryFocusNode);
```

Comment:
I do not understand the idea of the `autoFocus` logic 
It seems here that you want it to launch a first search, then focus the search_bar.
But : 
* In my point of view, auto focusing a search bar should not launch a search, just put the cursor in the bar
* As it is the first launch, `_searchQueryController.text` is empty
* Why don't you simply use the `autoFocus` field on TextField ?